### PR TITLE
Add support for . in attr types

### PR DESCRIPTION
**Pull request by frmdstryr**
I have not thoroughly tested this also I'm not sure if this covers the edge case(s) mentioned in https://github.com/nucleic/enaml/issues/351#issuecomment-508802683

### CODE REVIEW

**Comment by MatthieuDartiailh** [comment]
@sccolbert did you remember the edge case you mentioned ?

File: enaml/core/compiler_common.py [hunk 1]
```diff
@@ -259,6 +259,11 @@ def load_name(cg, name, local_names):
     """
     if name in local_names:
         cg.load_fast(name)
+    elif '.' in name:
+        parts = name.split('.')
+        load_name(cg, parts.pop(0), local_names)
+        while parts:
+            cg.load_attr(parts.pop(0))
```
Comment (sccolbert) [inline enaml/core/compiler_common.py]: This is the wrong place for this logic. The `name` argument is assumed to not have a dot. We should really make a new ast node to describe the dotted  type name, then do a more granular emit in the compiler.

Hiding the logic away in this function, while easy, is a bit too much magic for my taste.

**Comment by MatthieuDartiailh** [comment]
This was not added upfront to avoid people being bitten by the fact that the type should only reference globally accessible objects since it is accessed during the 'class' definition and not its instantiation.
Currently people can already end up in trouble is they do this:
```
enamldef Main(Window):
    attr a
    attr b : a = 1
```

We will have to clearly document this behavior.

**Comment by frmdstryr** [comment]
Thanks for the feedback. Instead of a separate ast node here would it be better to reuse the `dotted_name` and [ast_for_dotted_name](https://github.com/nucleic/enaml/blob/master/enaml/core/parser/base_parser.py#L51)?

**Comment by sccolbert** [comment]
> Thanks for the feedback. Instead of a separate ast node here would it be better to reuse the `dotted_name` and [ast_for_dotted_name](https://github.com/nucleic/enaml/blob/master/enaml/core/parser/base_parser.py#L51)?

Yes. And then update the compiler to handle that AST node in the type position.

File: enaml/core/parser/base_parser.py [hunk 2]
```diff
@@ -589,18 +593,21 @@ def p_storage_expr3(self, p):
...
         node.kind = kind
         node.name = p[2]
-        node.typename = p[4]
+        typename = ast_for_dotted_name(p[4])
+        typename.lineno = lineno
+        node.typename = typename
         node.expr = p[5]
         p[0] = node
 
+
```
Comment (MatthieuDartiailh) [inline enaml/core/parser/base_parser.py]: extra blank line

**Comment by MatthieuDartiailh** [comment]
ping @sccolbert @frmdstryr

**Comment by MatthieuDartiailh** [comment]
Ping @sccolbert

**Comment by MatthieuDartiailh** [comment]
ping @sccolbert

**Comment by sccolbert** [comment]
LGTM

**Comment by MatthieuDartiailh** [comment]
Thanks @sccolbert I will merge on my next spare cycle.

**Pull request merged**